CMakeLists.txt

```diff
@@ -711,6 +711,11 @@ function(build_tests)
       configure_file(EXAMPLES/MATRIX_MARKET/issue215.mtx ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/issue215.mtx)
       configure_file(EXAMPLES/MATRIX_MARKET/issue215.sh ${CMAKE_RUNTIME_OUTPUT_DIRECTORY}/issue215.sh)
       add_test(NAME issue215_tst WORKING_DIRECTORY ${CMAKE_RUNTIME_OUTPUT_DIRECTORY} COMMAND ${BASH_PROGRAM} -eu issue215.sh)
+      if (WIN32 AND BUILD_SHARED_LIBS)
+        set_tests_properties(arpackmm_tst issue401_tst issue215_tst
```

Comment:
Could you add a cmake `function(add_test_with_working_dir test_name)` to gather `add_test` and `set_tests_properties` for the 3 tests that need a working dir? Would be clearer I guess

Reply:
I added that function in the last commit.

Is that what you had in mind?

Reply:
> Is that what you had in mind?

Yes!